### PR TITLE
fixes #14162 - hide unsupported sub-commands and options

### PR DESCRIPTION
**Pull request by thomasmckay**
This PR is intended to hide everything except settings import/export as the start of fully supporting this in both the upstream and downstream products.

+ updated for ruby 2.0.0
+ updated README
+ fixed rubocop

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,14 +2,17 @@ source "https://rubygems.org"
 
 gemspec
 
+# for generating i18n files, gettext > 3.0 dropped ruby 1.8 support
```
Comment (komidore64) [inline Gemfile]: i didn't think we supported 1.8 anymore?

Comment (thomasmckay) [inline Gemfile]: The comment was brought over from katello's Gemfile. Foreman's has the same line but no comment. Suggestions?

Comment (komidore64) [inline Gemfile]: I think as of Katello 2.4, hammer has also been running inside of the ruby1.9.3 scl. Let me double check.

Comment (komidore64) [inline Gemfile]: Yeah, hammer runs inside of the ``tfm`` SCL which is ruby 1.9.3.

```
# scl enable tfm "ruby --version"
ruby 1.9.3p484 (2013-11-22 revision 43786) [x86_64-linux]
```

But, I guess if hammer-cli-katello still requires it then it's fine.

Question: When generating translation files, does the rake task only run inside of hammer-cli-csv? Or does a gettext require an entire loaded hammer environment with all modules?

Comment (thomasmckay) [inline Gemfile]: TBH, I haven't investigated translation yet. It's on the list to figure out based on other hammer repos. Would you rather it be in this PR or a separate?

Comment (komidore64) [inline Gemfile]: If that's the case, then I'd say leave it out for now.

Comment (thomasmckay) [inline Gemfile]: Good point, removed.

File: lib/hammer_cli_csv/csv.rb [hunk 2]
```diff
@@ -3,9 +3,59 @@
...
+                subcommand_class == HammerCLICsv::CsvCommand::RolesCommand ||
+                subcommand_class == HammerCLICsv::CsvCommand::SmartProxiesCommand ||
+                subcommand_class == HammerCLICsv::CsvCommand::SpliceCommand ||
+                subcommand_class == HammerCLICsv::CsvCommand::SubnetsCommand ||
+                subcommand_class == HammerCLICsv::CsvCommand::SubscriptionsCommand ||
+                subcommand_class == HammerCLICsv::CsvCommand::SyncPlansCommand ||
+                subcommand_class == HammerCLICsv::CsvCommand::UsersCommand
+          else
+            false
+          end
```
Comment (komidore64) [inline lib/hammer_cli_csv/csv.rb]: What does this giant abomination do?

Comment (thomasmckay) [inline lib/hammer_cli_csv/csv.rb]: This excludes the subcommands that I do not wish to be listed in 'hammer csv --help' at this time. All the subcommands still exist but they are never shown.

**Comment by thomasmckay** [comment]
@komidore64 - thoughts?

File: lib/hammer_cli_csv/csv.rb (hunk 2 above)
Comment (komidore64) [inline lib/hammer_cli_csv/csv.rb]: Maybe it would be shorter to whitelist commands rather than blacklisting them?

Comment (thomasmckay) [inline lib/hammer_cli_csv/csv.rb]: That's true at the moment but the goal is to get the blacklisting shorter. :) Consider this f'ugly list my incentive.

Comment (beav) [inline lib/hammer_cli_csv/csv.rb]: Are newly added commands expected to work out of the box? If not, I would say a whitelist is better. If new stuff should "just work", then a blacklist works better.

Also, can this be changed to an array of class names? That way you could keep it up top in a constant and update as needed, then just check if `subcommand_class` exists in the array. I misread this block a couple of times and thought it was doing something totally different :evergreen_tree:

**Comment by thomasmckay** [comment]
@beav @komidore64 - I updated the logic to have commands default to unsupported. The supported commands are settings, export, and import. rake tests pass for those three.

**Comment by komidore64** [comment]
yeah that looks nicer

**Comment by komidore64** [comment]
ACK

**Pull request merged**